### PR TITLE
chore: first pass at maintainer covenant, fix #586

### LINKED ISSUE #586: Adopt and publish a "Maintainer Covenant" for this project
Are there any opensource projects that publish covenants around their security / maintenance practices? Something we could use as a template...?

I'm thinking of something similar in spirit to the [contributor convenant](https://www.contributor-covenant.org/), but that enumerates basic principles of good project management (esp. as relates to security) that the maintainers commit to.  I ask because there has been yet another [breach in a popular NPM module](https://github.com/faisalman/ua-parser-js/issues/536), and the circumstances seem all too familiar.   Having such a document would, I believe, help encourage maintainers to check these boxes as their projects become more popular, and I believe (read, "hope"), that we've reached a place where this project could set a good example.

Off the top of my head, some items this could cover:

* Security
  * Use of 2FA (required for all maintainers, all relevant accounts.  Esp. GitHub & NPM)
  * Use of password managers for managing credentials on personal devices
  * Handling of shared passwords and access tokens (ref. [Secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets))
* Maintainer team
  * Minimum size (to allow for code review, and mitigate [Bus Factor](https://en.wikipedia.org/wiki/Bus_factor) issues)
  * Qualifications expected
  * Vetting process
* Code access
  * All commits to master require review

### PR DESCRIPTION
**Pull request by broofa**
First pass at a Maintainer Covenant.  This is pretty rough, but let's get this to a form where it's helpful to people and then we can iterate.

Goals:

* Primary: Help users answer the question, "Can this project be trusted?"
* Secondary: Document the steps we have and, more importantly, have not taken to insure the integrity of this project
* Secondary: Provide inspiration and a template for other projects to do the same

### A note about "criteria"

I found myself drawing inspiration from ESLint, of all things.  I felt it was important to have specific, measurable (or at least evaluatable) criteria by which to assess a project.  Giving these ESLint-like names (e.g. `no-shared-accounts`) makes it easy to refer to and discuss each specific point.  It also gives maintainers a way of easily enumerating what it is they do / don't comply with.

The criteria I've listed so far are pretty specific to security, but I can imagine other stuff outside of that.  E.g. `unit-tests` or `typed-language` that speak more to code quality.  A part of me thinks this would be useful, but another part of me feels that sort of thing should be out of scope.  Thoughts?

[Note: I accidentally pushed this to `master` then had to revert... and added the "no-main-commit" criterion as a result. 😂 ]

### CODE REVIEW

**Comment by broofa** [comment]
News: `coa` and `rc` packages compromised [source](https://www.bleepingcomputer.com/news/security/popular-coa-npm-library-hijacked-to-steal-user-passwords/)

**Comment by TrySound** [comment]
Funny thing, I maintain svgo, which depended on coa in v1 which I deprecated recently.
v1 still has 1M downloads. Interesting if anybody will upgrade.

File: MAINTAINER_COVENANT.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+The maintainers of this project are:
+
+|                              |                           |
+| ---------------------------- | ------------------------- |
+| **Christoph Tavan, @ctavan** | Google                    |
+| **Robert Kieffer, @broofa**  | Google, Facebook, CodePen |
+
+**Criteria**
+
+- [x] `bus-factor`: To mitigate the "[bus factor](https://en.wikipedia.org/wiki/Bus_factor)" risk, projects are expected to have 2 or more maintainers
```
[the remark this answers is not in the document]
Comment (broofa) [reply MAINTAINER_COVENANT.md:34]: I believe [Bus factor](https://en.wikipedia.org/wiki/Bus_factor) is the more popular term of art.  It also appeals to m slightly macabre sense of humor.

**Comment by broofa** [comment]
Closing this, as I think there's a better approach (basically a separate site that projects can link to) and I'd rather not complicate this project further.

**Pull request closed without merging**